lms/static/js/spec/dashboard/track_events_spec.js

```diff
@@ -14,70 +14,51 @@
+                                           window.edx.dashboard.generateTrackProperties];
 
-            it('sends an analytics event when the user clicks enter course link', function() {
-                // Verify that analytics events fire when the options dropdown is engaged.
-                expect(window.analytics.trackLink).toHaveBeenCalledWith(
-                    $('.wrapper-action-more'),
-                    'edx.bi.dashboard.course_options_dropdown.clicked',
-                    window.edx.dashboard.generateTrackProperties
-                );
-            });
+                var courseOptionDropdownClickedArgs = [$('.wrapper-action-more'),
+                                                       'edx.bi.dashboard.course_options_dropdown.clicked',
+                                                       window.edx.dashboard.generateTrackProperties];
 
-            it('sends an analytics event when the user clicks the learned about verified track link', function() {
-                //Verify that analytics events fire when the 'Learned about verified track' link is clicked.
-                expect(window.analytics.trackLink).toHaveBeenCalledWith(
-                    $('.verified-info'),
-                    'edx.bi.dashboard.verified_info_link.clicked',
-                    window.edx.dashboard.generateTrackProperties
-                );
-            });
+                var verifiedInfoLinklArgs = [$('.verified-info'),
+                                             'edx.bi.dashboard.verified_info_link.clicked',
+                                             window.edx.dashboard.generateTrackProperties];
 
-            it('sends an analytics event when the user clicks find courses button', function() {
-                // Verify that analytics events fire when the 'user clicks find the course' button.
-                expect(window.analytics.trackLink).toHaveBeenCalledWith(
-                    $('.btn-find-courses'),
-                    'edx.bi.dashboard.find_courses_button.clicked',
-                    {
-                        category: 'dashboard',
-                        label: null
-                    }
-                );
-            });
+                var findCourseButtonArgs = [$('.btn-find-courses'),
+                                            'edx.bi.dashboard.find_courses_button.clicked',
+                                            {
+                                                category: 'dashboard',
+                                                label: null
+                                            }];
 
-            it('sends an analytics event when the user clicks the \'View XSeries Details\' button', function() {
-                expect(window.analytics.trackLink).toHaveBeenCalledWith(
-                    $('.xseries-action .btn'),
-                    'edx.bi.dashboard.xseries_cta_message.clicked',
-                    window.edx.dashboard.generateProgramProperties
-                );
```

Comment:
@cpennington Thank you for the closer look.
First, I believe we are using Jasmine 1.3 instead of Jasmine 2. Therefore the documentation should be http://jasmine.github.io/1.3/introduction.html#section-Spies
Nevertheless, the 1.3 documentation says the same that "The toHaveBeenCalledWith matcher will return true if the argument list matches any of the recorded calls to the spy."
This explains why I cannot reproduce the issue, and why the original test fails only a couple of times for the past few month.
However, if you eye ball the last error message I collected in the above comments, there should not be any reason for the test to fail.
At this point, I am also questioning what the root cause actually is. Do you think there will be enough data to figure out the root cause?